tactics/class_tactics.ml

```diff
@@ -1116,11 +1116,24 @@ let evar_dependencies pred evm p =
 
 (** [split_evars] returns groups of undefined evars according to dependencies *)
 
-let split_evars pred evm =
+let split_evars pred env evm =
   let p = Intpart.create () in
   evar_dependencies pred evm p;
   deps_of_constraints (snd (extract_all_conv_pbs evm)) evm p;
-  Intpart.partition p
+  let part = Intpart.partition p in
```

Comment:
Doing the strictly_unique analysis after the partition means 2 evars could be connected through a strictly unique evar `?a <-> ?strictly_unique <-> ?b` and stay connected after it is removed from the component.
Is that what we want?

Reply:
Hm, that's an interesting question. I did it this way because it was a tiny code change and my understanding of the remaining code is very superficial. It seems fine to me disconnect `?a` and `?b` in your example and variables like them in general as long as they are scheduled before the strictly unique groups at the end.

Reply:
cf pushed commit
NB we should have some tests for this